x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { useState, useCallback } from 'react';
+import { Moment } from 'moment';
+
+interface Props {
+  setupModule: (startTime?: number | undefined, endTime?: number | undefined) => void;
+  retrySetup: (startTime?: number | undefined, endTime?: number | undefined) => void;
+}
+function selectedDateToParam(selectedDate: Moment | null) {
+  if (selectedDate) {
+    return selectedDate.valueOf(); // To ms unix timestamp
+  }
+  return undefined;
+}
+
+export const useAnalysisSetupState = ({ setupModule, retrySetup }: Props) => {
+  const [startTime, setStartTime] = useState<Moment | null>(null);
```

Comment:
I can't see it would be a hassle, the implementation was copied over from the original `AnalysisSetupTimerangeForm` to this state. I agree with the objections r.e. Moment.